src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -13,45 +12,17 @@ public abstract class OwnedBuffer<T> : IDisposable, IRetainable
 
         public abstract int Length { get; }
 
-        public abstract Span<T> Span { get; }
+        public abstract Span<T> AsSpan(int index, int length);
```

Comment:
This is faster. The method is virtual, so the same optimizations we were able to do in Span probably will not work here. Moreover, it allows subclasses to materialize only part of the buffer representing this range.